### ISSUE TITLE
Kubernetes federation invalid reference format

### ISSUE THREAD

**Issue by Thermi** [kubernetes/kubernetes#53430]
/kind bug

**What happened**:
Federation set up using kubefed uses invalid reference format, breaking the Federation control plane.

```
r}   Warning   InspectFailed   kubelet, gke-kubernetes-cluster-pool-1-76a09ebd-b587   Failed to apply default image tag "gcr.io/google_containers/hyperkube-amd64:v1.8.0+0b9efaeb34a2f": couldn't parse image reference "gcr.io/google_containers/hyperkube-amd64:v1.8.0+0b9efaeb34a2f": invalid reference format
2017-10-04 14:07:14 +0200 CEST   2017-10-04 14:00:47 +0200 CEST   31        fed-zone-europe-controller-manager-4130961042-m9c3z.14ea5c3e2a63466e   Pod                 Warning   FailedSync   kubelet, gke-kubernetes-cluster-pool-1-76a09ebd-b587   Error syncing pod
```
**What you expected to happen**:
That it works.

**How to reproduce it (as minimally and precisely as possible)**:
Try to set up a control plane using kubefed.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 
Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.0+0b9efaeb34a2f", GitCommit:"638b0a06d928e7a12fa29bb12610158a235585d7", GitTreeState:"dirty", BuildDate:"2017-10-02T09:48:25Z", GoVersion:"go1.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"7+", GitVersion:"v1.7.6-gke.1", GitCommit:"407dbfe965f3de06b332cc22d2eb1ca07fb4d3fb", GitTreeState:"clean", BuildDate:"2017-09-27T21:21:34Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration**:
GCP
- OS (e.g. from /etc/os-release):
Ubuntu Xenial
- Kernel (e.g. `uname -a`): Irrelevant
- Install tools: GKE

**Comment by quinton-hoole** [comment]
cc @madhusudancs @shashidharatd might be able to help.

**Comment by matchstick** [comment]
@shashidharatd  Assigning to you from the build cop rotation.

**Comment by shashidharatd** [comment]
@Thermi, the fix for this issue is identified and fixed in #53482. But it will take a while to be available on stable releases. Meanwhile you can use `--image` flag to `kubefed init` to pass specific hyperkube version.
